Clother/View Models/ClothesViewModel.swift

```diff
@@ -26,59 +26,120 @@ class ClothesViewModel {
     }
     
     // TODO: Creates a clothing item and stores it in our application.
     func createClothing(name: String, image: Data, type: Clothing.Kind, size: Clothing.Size, color: Clothing.Color, weather: Clothing.Weather, fabric: Clothing.Fabric) {
-        
+        let clothing = ClothingItem(name: name, type: type, size: size, color: color, weather: weather, fabric: fabric, displayImage: image)
+        userClothes.append(clothing)
     }
     
     /* TODO: Randomly creates an outfit out of the provided constraints and returns it. Do not store it anywhere.
-        Instead of an OutfitItem, you will return a tuple of 3 values in the form of (top, bottom, shoes, accessories). If no top/bottom/shoes are available on the app, return nil.
-        If a provided constraint is nil, it will not be enforced. HINT: you can utilize createOutfit(...) to avoid repeating code.
-        NOTE: leave the name to be empty.
+     Instead of an OutfitItem, you will return a tuple of 3 values in the form of (top, bottom, shoes, accessories). If no top/bottom/shoes are available on the app, return nil.
+     If a provided constraint is nil, it will not be enforced. HINT: you can utilize createOutfit(...) to avoid repeating code.
+     NOTE: leave the name to be empty.
      */
-    func generateRandomOutfit(type: Clothing.Kind?, size: Clothing.Size?, color: Clothing.Color?, weather: Clothing.Weather?, fabric: Clothing.Fabric?)
+    func generateRandomOutfit(size: Clothing.Size?, color: Clothing.Color?, weather: Clothing.Weather?, fabric: Clothing.Fabric?)
     -> (ClothingItem, ClothingItem, ClothingItem, [ClothingItem])? {
-        // placeholder
-        return (.sample, .sample, .sample, [.sample])
+        var filteredTops: [ClothingItem] = []
+        var filteredBottoms: [ClothingItem] = []
+        var filteredShoes: [ClothingItem] = []
+        var filteredAccessories: [ClothingItem] = []
+        
+        for item in userClothes {
+            if let size = size, item.size != size {
+                continue
+            }
+            if let color = color, item.color != color {
+                continue
+            }
+            if let weather = weather, item.weather != weather {
+                continue
+            }
+            if let fabric = fabric, item.fabric != fabric {
+                continue
+            }
+            
+            switch item.type {
+            case .top:
+                filteredTops.append(item)
+            case .bottom:
+                filteredBottoms.append(item)
+            case .shoes:
+                filteredShoes.append(item)
+            case .accessories:
+                filteredAccessories.append(item)
+            default:
+                continue
+            }
+        }
+        guard !filteredTops.isEmpty, !filteredBottoms.isEmpty, !filteredShoes.isEmpty, !filteredAccessories.isEmpty else {
+            return nil
+        }
+        let randomTop = filteredTops.randomElement()!
```

Comment:
Don't user force unwrapping, we try to never use it in our code because we never know when it might throw an error. A more safe of doing what you just did would be to do guard let on the random elements. Then return nil if one of them was nil.